### PR TITLE
fix: enable cookie propagation gating in local, pr, dev

### PR DESCRIPTION
**Pull request by dszheng92**
Ticket: BG-63404

### CODE REVIEW

File: modules/sdk-api/src/bitgoAPI.ts [hunk 1]
```diff
@@ -187,10 +187,7 @@ export class BitGoAPI implements BitGoBase {
         common.Environments[env].stellarFederationServerUrl = params.stellarFederationServerUrl;
       }
       if (
-        params.customRootURI &&
-        params.customRootURI !== Environments.prod.uri &&
-        params.customRootURI !== Environments.test.uri &&
-        params.cookiesPropagationEnabled
+        params.customRootURI && params.cookiesPropagationEnabled
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-api/src/bitgoAPI.ts:190]: Let's migrate the tests from `bitgo` to `sdk-api` and update them to match this change.

**Review by vitalir-bg — COMMENTED** [review]
Looks good to me but lint check seems to be failing

File: modules/sdk-api/src/bitgoAPI.ts [hunk 2]
```diff
@@ -187,10 +187,7 @@ export class BitGoAPI implements BitGoBase {
         common.Environments[env].stellarFederationServerUrl = params.stellarFederationServerUrl;
       }
       if (
-        params.customRootURI &&
-        params.customRootURI !== Environments.prod.uri &&
-        params.customRootURI !== Environments.test.uri &&
```
[the remark this answers is not in the document]
Comment (dszheng92) [reply modules/sdk-api/src/bitgoAPI.ts:192]: Environments import is removed

File: modules/sdk-api/src/bitgoAPI.ts (hunk 1 above)
Comment (dszheng92) [reply modules/sdk-api/src/bitgoAPI.ts]: moved this test to sdk-api/test/unit

File: modules/sdk-api/test/unit/bitgoAPI.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import 'should';
+import * as BitGoJS from '../../../bitgo/src';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-api/test/unit/bitgoAPI.ts:2]: See wallet.ts test for using `BitGoAPI` over `BitGoJS`.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Review by vitalir-bg — APPROVED** [review]
The changes look good to me but I am not quite sure why are they necessary.

It looks like the purpose is ultimately to allow the SDK to make calls to Lumina backend but I am not sure if the SDK should make these calls.

**Pull request merged**